bql/grammar/grammar_test.go

```diff
@@ -145,6 +145,18 @@ func TestAcceptByParse(t *testing.T) {
 		 from ?b where {?s "old_predicate_1"@[,] ?o1.
 				?s "old_predicate_2"@[,] ?o2.
 				?s "old_predicate_3"@[,] ?o3};`,
+		// Test Deconstruct clause.
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o} having ?s = ?o;`,
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o};`,
+		`deconstruct {?s ?p ?o.
+			      _:v "_subject"@[] ?s.
```

Comment:
Mmh, I recall we discussed to start with the triple syntax with not reification. I thought that would include not supporting blank node syntax for deconstruct right now. Also not sure what the value for those would be since those are uniquely generated, likely it would not deconstruct anything.